ws-server/src/server.rs

```diff
@@ -169,19 +173,27 @@ async fn background_task(
 	let (mut sender, mut receiver) = server.into_builder().finish();
 	let (tx, mut rx) = mpsc::unbounded::<String>();
 
+	let conn = is_conn.clone();
 	// Send results back to the client.
 	tokio::spawn(async move {
-		while let Some(response) = rx.next().await {
-			log::debug!("send: {}", response);
-			let _ = sender.send_text(response).await;
-			let _ = sender.flush().await;
+		while conn.load(std::sync::atomic::Ordering::SeqCst) {
+			match rx.next().await {
+				Some(response) => {
+					log::debug!("send: {}", response);
+					let _ = sender.send_text(response).await;
+					let _ = sender.flush().await;
+				}
+				None => break,
+			};
 		}
+		// terminate connection.
+		let _ = sender.close().await;
 	});
 
 	// Buffer for incoming data.
 	let mut data = Vec::with_capacity(100);
 
-	loop {
+	while is_conn.load(std::sync::atomic::Ordering::SeqCst) {
```

Comment:
Why `SeqCst`? Are ordering rules important here?
IMHO `Relaxed` will do just fine.

Reply:
I have screwed up atomic ordering so bad so for now I use strongest order by default.

We can fine-tune it's too slow

Reply:
Makes sense, why not :)